packages/amplify-graphql-transformer-core/src/migration/migrate.ts

```diff
@@ -0,0 +1,144 @@
+import * as fs from 'fs-extra';
+import { parse, print, visit } from 'graphql'
+import { migrateKeys } from './migrators/key'
+import { migrateAuth } from './migrators/auth'
+import { migrateConnection } from './migrators/connection'
+import {
+  backupSchema,
+  getDefaultAuthFromContext,
+  readSchemaDocuments,
+  readSingleFileSchema,
+  replaceFile,
+  SchemaDocument,
+  undoAllSchemaMigration,
+} from './utils';
+import { DocumentNode } from 'graphql/language';
+import { prompter, printer } from 'amplify-prompts';
+import path from 'path';
+import { $TSContext, exitOnNextTick, FeatureFlags, pathManager, stateManager } from 'amplify-cli-core';
+
+const cliToMigratorAuthMap: Map<string, string> = new Map<string, string>([
+  ['API_KEY', 'apiKey'],
+  ['AWS_IAM', 'iam'],
+  ['AMAZON_COGNITO_USER_POOLS', 'userPools'],
+  ['OPENID_CONNECT', 'oidc']
+]);
+
+function migrateGraphQLSchema(schema: string, authMode: string, massSchema: DocumentNode): string {
+  let output = parse(schema)
+  visit(output, {
+    ObjectTypeDefinition: {
+      enter(node) {
+        migrateKeys(node);
+        migrateAuth(node, authMode);
+        migrateConnection(node, massSchema);
+        return node;
+      }
+    }
+  });
+
+  return print(output);
+}
+
+async function askMigration(singleFileExists: boolean, directoryExists: boolean): Promise<boolean> {
+  await printer.info(`About to begin schema migration, found valid target(s) for migration:\n${singleFileExists ? 'schema.graphql\n' : ''}${directoryExists ? 'schema directory\n' : ''}\n`);
+  await printer.info('If you confirm migration, you will receive no further prompts and all migration will be completed. The CLI will exit after migration and will not push your resources to give you time to review.');
+  await printer.warn('If you are using @auth, we strongly recommend spending extra time reviewing your auth rules after migration to ensure they\'re accurate');
+  return await prompter.confirmContinue("About to start migrating to V2 transformer, are you ready to continue?");
+}
+
+async function runMigration(schemas: SchemaDocument[], authMode: string): Promise<void> {
+  let fullSchema: string;
+  let schemaList: string[] = new Array(schemas.length);
+  let backupSchemaPromiseArray: Promise<void>[] = [];
+  schemas.forEach((doc, idx) => {
+    schemaList[idx] = doc.schema;
+    backupSchemaPromiseArray.push(backupSchema(doc.filePath));
+  });
+  for (let promise of backupSchemaPromiseArray) {
+    await promise;
+  }
```

Comment:
You probably want `Promise.all()` or `Promise.allSettled()` here.

Reply:
yes I do, thanks!